okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -402,16 +371,15 @@ public void cancel() {
   }
 
   public void streamFailed(@Nullable IOException e) {
-    Socket socket;
-    Connection releasedConnection;
-
     synchronized (connectionPool) {
+      if (released) throw new IllegalStateException();
```

Comment:
Won't this likely hide the triggering exception?

Reply:
I should have used AssertionError. It's just a sanity check to confirm this is called before the connection is released.